### PR TITLE
Implement XCFocus doc / Next steps for XCHammer

### PR DESCRIPTION
**Pull request by jerrymarino**

### CODE REVIEW

File: Docs/XCHammerFocus.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+- easily switch between `Focus` and non `Focus`. Build results are shared
+  through Bazel by default.
+
+### Known issues
+
+- Xcode progress bar doesn't work ( see future research )
+
+- Indexing can have issues / be invalidated after builds  TODO: Add github issue
+
+### Future research / Roadmap
```
Comment (jerrymarino) [inline Docs/XCHammerFocus.md]: One of the things that we're going to need in XCHammer to be more data driven around the local development experience is instrumentation. We'll need to instrument generation time, local build time and XCHammer is going to need some hooks for this.

File: Docs/XCHammerFocus.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+- Fast bisecting / Top Hatting / PR Rewiew: The Xcode project downloads
+  pre-built artifacts from CI pipelines to make testing PRs and bisecting
+  easier. _CI integration ends up in either the XCHammer repostiory or external
+  repositories to open source workflows for Buildkite or other CI systems._
+
+- Project caching: XCHammer downloads pre-compiled projects from CI and
```
Comment (mikezucc) [inline Docs/XCHammerFocus.md]: +1 for speeding up branch switches

**Comment by mikezucc** [comment]
i like this roadmap!

File: Docs/XCHammerFocus.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+XCHammer project.
+
+## Preface
+
+The large size of the Pinterest iOS code base has exceeded the limits of
+efficient operation in a normal Xcode project. The point of XCHammer was
+originally to produce a pure Xcode project.
+
+Generally, a large Xcode project is required to index and compile all of the
+sources. When the project gets large, e.g. at Pinterest, the cost of indexing
```
[the remark this answers is not in the document]
Comment (jerrymarino) [reply Docs/XCHammerFocus.md]: Done

File: Docs/XCHammerFocus.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+and building grows, and ends up slowing down development. XCFocus improves the
+situation by removing unnecessary source files from Xcode and relies heavily on
```
[the remark this answers is not in the document]
Comment (jerrymarino) [reply Docs/XCHammerFocus.md]: Chanced to uncommon

File: Docs/XCHammerFocus.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+Bazel for caching and faster builds. 
+
+### Goals and differences
+
+- Less source files: project generation time is several times faster.
```
[the remark this answers is not in the document]
Comment (jerrymarino) [reply Docs/XCHammerFocus.md]: Done

File: Docs/XCHammerFocus.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+- Indefinite caching: build artifacts are cached across builds, so branch
+  switching is faster. e.g. cocoapod dependencies are rarely rebuilt on device,
+  and there is no additional system required to achieve this.
+
+- Optional generation: the Xcode project doesn't have to be regenerated and will
```
[the remark this answers is not in the document]
Comment (jerrymarino) [reply Docs/XCHammerFocus.md]: Done

**Pull request merged**